crates/biome_service/src/file_handlers/html.rs

```diff
@@ -450,6 +465,74 @@ fn format(
     }
 }
 
+fn format_embedded(
+    biome_path: &BiomePath,
+    document_file_source: &DocumentFileSource,
+    parse: AnyParse,
+    settings: &Settings,
+    embedded_nodes: Vec<FormatEmbedNode>,
+) -> Result<Printed, WorkspaceError> {
+    let options = settings.format_options::<HtmlLanguage>(biome_path, document_file_source);
+
+    let tree = parse.syntax();
+    let indent_script_and_style = options.indent_script_and_style().value();
+    let mut formatted = format_node(options, &tree)?;
+    let new_document = formatted.format_embedded(move |range| {
+        let mut iter = embedded_nodes.iter();
+        let node = iter.find(|node| node.range == range)?;
```

Comment:
Would it be possible to identify the node with an index instead of a range? It seems unfortunate we need to iterate all nodes again just to compare ranges.